Libraries/Core/Timers/JSTimers.js

```diff
@@ -131,14 +132,24 @@ const JSTimers = {
   /**
    * @param {function} func Callback to be invoked every frame and provided
    * with time remaining in frame.
+   * @param {?object} options
    */
-  requestIdleCallback: function(func : Function) {
+  requestIdleCallback: function(func : Function, options : ?Object) {
     if (JSTimersExecution.requestIdleCallbacks.length === 0) {
       RCTTiming.setSendIdleEvents(true);
     }
 
     const id = _allocateCallback(func, 'requestIdleCallback');
     JSTimersExecution.requestIdleCallbacks.push(id);
+    if (options && options.timeout) {
+      const currentTime = performanceNow();
+      JSTimers.setTimeout(() => {
```

Comment:
We'll need to cancel this timeout if the idle callback is called before it triggers.